pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,384 @@
+    return strict_discrete_range(value, values, 4)
+
+
+SAMPLE_COUNT_VALUES = [8, 128e6]
+PRE_TRIG_VALUES = [0, 128e6]
+RECORD_COUNT_VALUES = [1, 1024]
+
+TRIGGER_HOLDOFF_VALUES = ureg.Quantity(np.asarray([0, 10]), ureg.s)
+TRIGGER_DELAY_VALUES = ureg.Quantity([0, 3600], ureg.s)
+
+
+def _get_channel_config_process(values):
+    result = {
+        'range': ureg.Quantity(Decimal(values[0]), ureg.V),
+        'coupling': values[1],
+        'filter': values[2]
+    }
+    return result
+
+
+def _get_acq_config_process(values):
+    return {
+        'sample_rate': ureg.Quantity(values[0], ureg.Hz),
+        'samples_per_record': int(values[1]),
+        'pre_trig_samples': int(values[2]),
+        'num_records': int(values[3]),
+        'trigger_holdoff': ureg.Quantity(values[4], ureg.s),
+        'trigger_delay': ureg.Quantity(values[5], ureg.s)
+    }
+
+
+def _set_acq_config_process(value):
+    return '{},{},{},{},{},{}'.format(
+        assume_units(value['sample_rate'], ureg.Hz).m_as(ureg.Hz),
+        value['samples_per_record'],
+        value['pre_trig_samples'],
+        value['num_records'],
+        assume_units(value['trigger_holdoff'], ureg.s).m_as(ureg.s),
+        assume_units(value['trigger_delay'], ureg.s).m_as(ureg.s))
+
+
+VOLTAGE_RANGE_VALUES = ureg.Quantity(np.asarray([0.25, 0.5, 1, 2, 4, 8, 16, 32, 128, 256]), ureg.V)
+
+COUPLING_VALUES = ['DC', 'AC']
+FILTER_VALUES = ['LP_20_MHZ', 'LP_2_MHZ', 'LP_200_KHZ']
+
+SAMPLE_RATE_VALUES = ureg.Quantity(
+    np.asarray([
+        1000, 2000, 5000, 10000, 20000, 50000,
+        100000, 200000, 500000, 1000000, 2000000, 5000000, 10000000, 20000000
+    ]), ureg.Hz)
+
+
+def _validate_acq_config(value, values):
+    strict_discrete_set(value['sample_rate'], AgilentL4534A.SAMPLE_RATE_VALUES)
+    strict_discrete_range(value['samples_per_record'], SAMPLE_COUNT_VALUES, 4)
+    strict_discrete_range(value['pre_trig_samples'], PRE_TRIG_VALUES, 4)
+    strict_range(value['num_records'], RECORD_COUNT_VALUES)
+    strict_range(value['trigger_holdoff'], TRIGGER_HOLDOFF_VALUES)
+    strict_range(value['trigger_delay'], TRIGGER_DELAY_VALUES)
+    return value
+
+
+def _validate_channel_config(value, values):
+    if 'range' not in value:
+        raise ValueError('Requires \'range\'')
+    if 'coupling' not in value:
+        raise ValueError('Requires \'coupling\'')
+    if 'filter' not in value:
+        raise ValueError('Requires \'filter\'')
+
+    strict_discrete_set(value['range'], VOLTAGE_RANGE_VALUES)
+    strict_discrete_set(value['coupling'], COUPLING_VALUES)
+    strict_discrete_set(value['filter'], FILTER_VALUES)
+    return value
+
+
+class AgilentL4534A(Instrument):
+    """
+    Represents the Agilent L4532A/L4534A digitizers.
+    """
+    def __init__(self, adapter, name="Agilent L4534A Digitizer", **kwargs):
+        super().__init__(
+            adapter, name, **kwargs
+        )
+
+    SAMPLE_RATE_VALUES = SAMPLE_RATE_VALUES
+
+    class DigitizerChannel(Channel):
```

Comment:
I'm not familiar with Agilient digitizers, but would other models in that product line be able to use `DigitizerChannel`? Maybe with dynamic properties? If so, I think it'd make more sense to move `DigitizerChannel` out of the `AgilentL4534A` namespace and have it defined somewhere else. If not, it makes sense to leave it there.